File: prime/fsharp_prime/prime.fs
#light
open System

//comment
(* commment *)

let isPrime n =
  let sqrt' = (float >> sqrt >> int) n // square root of integer
  [ 2 .. sqrt' ] // all numbers from 2 to sqrt'
  |> List.forall (fun x -> n % x <> 0) // no divisors

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    for idx_i in 1 .. 100 do
      printfn "isPrime(%A)=%A" idx_i (isPrime idx_i)
    0

File: prime/csharp_prime/prime.cs
using System;

public class Example {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit(1);
    }

    //Console.WriteLine( args[1] );
    //Console.WriteLine("reservation");
    for( ulong idx_i = 0; idx_i < 100; idx_i++ ) {
      Console.WriteLine("is_prime (" + idx_i + ") = " +  isPrime(idx_i));
    }
  }

  private static int isPrime(ulong number) {
    ulong idx_i;
    for ( idx_i=2; idx_i < number; idx_i++ ) {
        if (number % idx_i == 0 && idx_i != number) {
          return 0;
        }
    }
    return 1;
  }
}
